docs/doxygen/collectives.md

```diff
@@ -1,20 +1,169 @@
+
++ The UCC\_COLL\_TYPE\_FAN\_OUT operation is a synchronizing operation like
+UCC\_COLL\_TYPE\_FAN\_OUT. In this operation, the root participant sends a
+synchronizing message to all non-root participants and completes. The non-root
+participant completes once it receives a message from the root participant.
+
+
++ In the UCC\_COLL\_TYPE\_GATHER operation, each participant of the collective
+operation sends data to the root participant. All participants send the same
+amount of data (block\_size) to the root. The size of the block is
+“dt\_elem\_size * count”. The total amount of data received by the root is equal
+to block_size * num\_participants. Here, the “count” represents the number of
+data elements. The
+"dt_elem_size" represents the size of the data element in bytes. The
+"num_participants" represents the number of participants in the team. The data on
+the root is placed in the receive buffer ordered by the “ep” ordering. For
+example, if the participants’ endpoints are ordered as “ep\_a” to “ep\_n”, the
+data from the participant with ep_i is placed as an “ith” block on the receive
+buffer.
+
++ The UCC\_COLL\_TYPE\_ALLGATHER operation is similar to UCC\_COLL\_TYPE\_GATHER
+with one exception. Unlike in GATHER operation, the result is available at all
+participants’ receive buffer instead of only at the root participant.
+
+    Each participant sends the data of size "block_size" to all other participants
+in the collective operation. The size of the block is “dt\_elem\_size * count”.
+Here, the “count” represents the number of data elements. The "dt_elem_size"
+represents the size of the data element in bytes. The data on each participant
+is placed in the receive buffer ordered by the “ep” ordering. For example, if
+the participants’ endpoints are ordered as “ep\_a” to “ep\_n”, the data from the
+participant with ep_i is placed as an “ith” block on the receive buffer.
+
++ In the UCC\_COLL\_TYPE\_SCATTER operation, the root participant of the
+collective operation sends data to all other participants. It sends the same
+amount of data (block_size) to all participants. The size of the block
+(block_size) is “dt_elem_size * count”. The total amount of data sent by the
+root is equal to block_size * num\_participants. Here, the “count” represents the
+number of data elements. The "dt_elem_size" represents the size of the data
+element in bytes. The "num_participants" represents the number of participants in
+the team.
+
++ In the UCC\_COLL\_TYPE\_ALLTOALL collective operation, all participants
+exchange a fixed amount of the data. For a given participant, the size of data
+in src buffer is “size”, where size is dt\_elem\_size * count * num_participants.
+Here, the “count” represents the number of data elements. The "dt_elem_size"
```

Comment:
i think changing to "Here, the “count” represents the number of data elements per destination" would be more clear